adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -503,19 +600,80 @@ case class ReferenceRegion(
   }
 
   /**
+   * Determines if two regions are colocated on the same strand.
+   *
+   * @param other The other region.
+   * @return True if the two are on the same strand, false otherwise
+   */
+  @inline final def sameStrand(other: ReferenceRegion): Boolean = {
+    strand == other.strand
+  }
+
+  /**
+   * Determines if two regions are colocated on the same reference name.
+   *
+   * @param other The other region.
+   * @return True if the two are on the same reference name, false otherwise.
+   */
+  @inline final def sameReferenceName(other: ReferenceRegion): Boolean = {
+    referenceName == other.referenceName
+  }
+
+  /**
    * @return The length of this region in bases.
    */
   def length(): Long = {
     end - start
   }
 
+  /**
+   * Subtracts another region.
+   *
+   * Subtracting in this case removes the entire region and returns up to two
+   * new regions.
+   * @param other The region to subtract.
+   * @param requireStranded Whether or not to require other be on same strand.
+   * @return A list containing the regions resulting from the subtraction.
+   */
+  def subtract(other: ReferenceRegion, requireStranded: Boolean = false): List[ReferenceRegion] = {
+    val newRegionStrand =
+      if (requireStranded) {
+        require(overlaps(other), "Region $other is not overlapped by $this!")
+        strand
+      } else {
+        require(covers(other), s"Region $other is not covered by $this!")
+        Strand.INDEPENDENT
+      }
+    val first = if (other.start > start) {
+      List(ReferenceRegion(referenceName,
```

Comment:
Yeah, actually, just `List` -> `Iterable` these few ones. And probably `List()` -> `Iterable.empty`.